packages/eas-cli/src/commands/env/update.ts

```diff
@@ -0,0 +1,209 @@
+import { Flags } from '@oclif/core';
+import chalk from 'chalk';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import {
+  EASEnvironmentFlag,
+  EASNonInteractiveFlag,
+  EASVariableScopeFlag,
+  EASVariableVisibilityFlag,
+} from '../../commandUtils/flags';
+import { EnvironmentVariableScope } from '../../graphql/generated';
+import { EnvironmentVariableMutation } from '../../graphql/mutations/EnvironmentVariableMutation';
+import { EnvironmentVariablesQuery } from '../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../log';
+import {
+  getDisplayNameForProjectIdAsync,
+  getOwnerAccountForProjectIdAsync,
+} from '../../project/projectUtils';
+import { selectAsync } from '../../prompts';
+import { promptVariableEnvironmentAsync, promptVariableValueAsync } from '../../utils/prompts';
+
+export default class EnvironmentVariableUpdate extends EasCommand {
+  static override description =
+    'update an environment variable on the current project or owner account';
+
+  static override hidden = true;
+
+  static override flags = {
+    name: Flags.string({
+      description: 'Name of the variable',
+    }),
+    value: Flags.string({
+      description: 'Text value or the variable',
+    }),
+    link: Flags.boolean({
+      description: 'Link shared variable to the project',
+    }),
+    ...EASVariableVisibilityFlag,
+    ...EASVariableScopeFlag,
+    ...EASEnvironmentFlag,
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.Analytics,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    let {
+      flags: {
+        name,
+        value,
+        scope,
+        'non-interactive': nonInteractive,
+        environment,
+        link,
+        visibility,
+      },
+    } = await this.parse(EnvironmentVariableUpdate);
```

Comment:
we should validate that `name`, `environemnt`, ... are specified when in non-interactive mode